crates/bpf-filtering/src/test_suite.rs

```diff
@@ -226,6 +231,43 @@ fn uninteresting_processes_ignored() -> TestCase {
     })
 }
 
+// Make sure that when a process is attached to a target cgroup, it gets full interest
+fn cgroups_tracked() -> TestCase {
+    TestCase::new("cgroups_tracked", async {
+        let mut report = TestReport {
+            success: true,
+            lines: vec![],
+        };
+        let mut bpf = load_ebpf();
+        attach_raw_tracepoint(&mut bpf, "cgroup_attach_task");
+
+        let mut interest_map = InterestMap::load(&mut bpf, INTEREST_MAP_NAME).unwrap();
+        interest_map.clear().unwrap();
+
+        let pid = std::process::id() as i32;
+        interest_map
+            .set(Pid::from_raw(pid), NOT_INTERESTING)
+            .unwrap();
+
+        let mut target_cgroup_map = Map::<Cgroup, u8>::load(&mut bpf, "m_cgroup_rules").unwrap();
+        let cgroup = random_name("/cgroups_tracked");
+        let cgroup_path: Cgroup = cgroup.parse().unwrap();
+        target_cgroup_map.map.insert(cgroup_path, 0, 0).unwrap();
+
+        // Spawn a process inside the targetted cgroup
+        let (child_pid, _id) = fork_in_temp_cgroup(&cgroup[1..]);
+
+        if interest_map.0.map.get(&child_pid.as_raw(), 0).unwrap() != INTERESTING.as_raw() {
+            report
+                .lines
+                .push("✗ event for uninteresting process not skipped".to_string());
```

Comment:
this error message seems not correct

Reply:
I've added comments, let me know if it's clearer now (or if you'd rather rename something)